src/Helper/MyParcelCollection.php

```diff
@@ -172,11 +172,26 @@ public function addConsignment(AbstractConsignment $consignment): self
         if ($consignment->getApiKey() === null) {
             throw new MissingFieldException('First set the API key with setApiKey() before running addConsignment()');
         }
-
         $consignment->validate();
-
         $this->push($consignment);
 
+        // Check if all consignments have the same carrier
```

Comment:
The collection may contain consignments for different carriers and even consignments with different api keys.